src/compiler/transformers/esDecorators.ts

```diff
@@ -0,0 +1,2087 @@
+                    const esDecorateStatement = factory.createExpressionStatement(esDecorateExpression);
+                    setSourceMapRange(esDecorateStatement, moveRangePastDecorators(member));
+                    statements.push(esDecorateStatement);
+                }
+                else if (isPropertyDeclaration(member)) {
+                    initializersName = memberInfo.memberInitializersName ??= createHelperVariable(member, "initializers");
+
+                    if (isStatic(member)) {
+                        thisArg = classInfo.classThis;
+                    }
+
+                    let descriptor: Expression | undefined;
+                    if (isPrivateIdentifier(member.name) && hasAccessorModifier(member) && createDescriptor) {
+                       descriptor = createDescriptor(member, visitNodes(modifiers, node => isStaticModifier(node) ? undefined : node, isModifier));
+                       memberInfo.memberDescriptorName = descriptorName = createHelperVariable(member, "descriptor");
+                       descriptor = factory.createAssignment(descriptorName, descriptor);
+                    }
+
+                    // _static_field_initializers = __esDecorate(null, null, _static_member_decorators, { kind: "field", name: "...", static: true, private: ..., access: { ... } }, _staticExtraInitializers);
+                    // _field_initializers = __esDecorate(null, null, _member_decorators, { kind: "field", name: "...", static: false, private: ..., access: { ... } }, _instanceExtraInitializers);
+                    const esDecorateExpression = emitHelpers().createESDecorateHelper(
+                        isAutoAccessorPropertyDeclaration(member) ?
+                            factory.createThis() :
+                            factory.createNull(),
+                        descriptor ?? factory.createNull(),
+                        memberDecoratorsName,
+                        context,
+                        initializersName,
+                        extraInitializers);
+                    const esDecorateStatement = factory.createExpressionStatement(esDecorateExpression);
+                    setSourceMapRange(esDecorateStatement, moveRangePastDecorators(member));
+                    statements.push(esDecorateStatement);
+                }
+            }
+
+            if (name === undefined) {
+                enterName();
+                if (useNamedEvaluation) {
+                    ({ referencedName, name } = visitReferencedPropertyName(member.name));
+                }
+                else {
+                    name = visitPropertyName(member.name);
+                }
+                exitName();
+            }
+
+            if (!some(modifiers) && (isMethodDeclaration(member) || isPropertyDeclaration(member))) {
+                setEmitFlags(name, EmitFlags.NoLeadingComments);

```

Comment:
why set NoLeadingComments for members with no modifiers? Is it because decorators would emit before the comments?

Reply:
It's to avoid duplicate comments. I'll add a note.